staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -194,10 +194,13 @@ type Config struct {
 	// values below here are targets for removal
 	//===========================================================================
 
-	// PublicAddress is the IP address where members of the cluster (kubelet,
+	// AdvertiseAddress is the IP address where members of the cluster (kubelet,
 	// kube-proxy, services, etc.) can reach the GenericAPIServer.
 	// If nil or 0.0.0.0, the host's default interface will be used.
-	PublicAddress net.IP
+	AdvertiseAddress net.IP
```

Comment:
I really hate to be this nitpicky, but maybe if it's not going to have a port attached to it it should be called an IP, not an address? It's a little weird that this gets combined with the port and stored in an "external address".

Reply:
I noticed that and thought that ExternalAddress should be renamed. The authority is made up of an address and port according to https://tools.ietf.org/html/rfc3986#section-3.2 but ExternalAuthority might be confusing.